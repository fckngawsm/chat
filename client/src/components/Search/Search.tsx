import { useState } from "react";
import {
  RaduisSize,
  StyledIcon,
  StyledSearch,
  StyledSearchContainer,
} from "./styled";

interface SearchProps {
  placeholder: string;
  raduiusSize: RaduisSize;
  viewIcon?: boolean;
}

export const Search = ({
  placeholder,
  raduiusSize,
  viewIcon = true,
}: SearchProps) => {
  const [isFocused, setIsFocused] = useState(false);

  return (
    <StyledSearchContainer>
      {viewIcon && (
        <StyledIcon
          size={20}
          $isFocused={isFocused}
          data-testid="search-icon"
        />
      )}
      <StyledSearch
        placeholder={placeholder}
        $raduiusSize={raduiusSize}
        onFocus={() => setIsFocused(true)}
        onBlur={() => setIsFocused(false)}
        $isFocused={isFocused}
      />
    </StyledSearchContainer>
  );
};
