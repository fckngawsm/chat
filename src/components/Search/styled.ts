import { MagnifyingGlass } from "@phosphor-icons/react";
import styled from "styled-components";

export const StyledSearchContainer = styled.div`
  position: relative;
  width: 100%;
  display: flex;
  align-items: center;
`;

export const StyledIcon = styled(MagnifyingGlass)<{ isFocused: boolean }>`
  position: absolute;
  top: 50%;
  left: ${(props) => (props.isFocused ? "10px" : "42%")};
  transform: translateY(-50%);
  transform: ${(props) =>
    props.isFocused ? "translateY(-50%)" : "translate(-50%, -50%)"};
  color: #999999;
  transition: left 0.3s ease, transform 0.3s ease;
  pointer-events: none;
`;

export const StyledSearch = styled.input<{ isFocused: boolean }>`
  width: 100%;
  border-radius: 5px;
  padding: 10px 10px 10px ${(props) => (props.isFocused ? "32px" : "32px")};
  background-color: #efefef;
  color: #1e1e1e;
  font-size: 12px;
  font-weight: 500;
  font-family: "Inter", Arial, Helvetica, sans-serif;
  box-sizing: border-box;
  outline: none;
  border: none;
  height: 32px;
  text-align: center;
  caret-color: transparent;

  &:focus {
    outline: none;
    text-align: left;
    padding-left: 32px;
  }
  &:focus::placeholder {
    opacity: 0;
  }

  &::placeholder,
  ::-webkit-input-placeholder {
    color: #999999;
    font-size: 12px;
    font-weight: 500;
    font-family: "Inter", Arial, Helvetica, sans-serif;
  }
  :-ms-input-placeholder {
    color: #999999;
    font-size: 12px;
    font-weight: 500;
    font-family: "Inter", Arial, Helvetica, sans-serif;
  }
`;
