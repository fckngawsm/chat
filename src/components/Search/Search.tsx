import { useState } from "react";
import { StyledIcon, StyledSearch, StyledSearchContainer } from "./styled";
export const Search = () => {
  const [isFocused, setIsFocused] = useState(false);

  return (
    <StyledSearchContainer>
      <StyledIcon size={20} isFocused={isFocused} />
      <StyledSearch
        placeholder="Поиск"
        onFocus={() => setIsFocused(true)}
        onBlur={() => setIsFocused(false)}
        isFocused={isFocused}
      />
    </StyledSearchContainer>
  );
};
