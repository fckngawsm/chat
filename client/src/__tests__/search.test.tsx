import { render, screen } from "@testing-library/react";
import { Search } from "../components/Search/Search";

describe("correct render Search", () => {
  it("should render Search component", () => {
    render(
      <Search placeholder="Пейсхолдер!" raduiusSize="s" viewIcon={false} />
    );

    expect(screen.getByPlaceholderText("Пейсхолдер!")).toMatchSnapshot();
  });
});
