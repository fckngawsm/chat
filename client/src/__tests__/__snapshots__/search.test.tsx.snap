// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct render Search should render Search component 1`] = `
<input
  class="sc-eqUzNf gtrqmr"
  placeholder="Пейсхолдер!"
/>
`;
